charts/sonarqube/templates/sonarqube-sts.yaml

```diff
@@ -22,6 +31,32 @@ spec:
     matchLabels:
       app: {{ template "sonarqube.name" . }}
       release: {{ .Release.Name }}
+{{- if and .Values.persistence.enabled (not .Values.persistence.existingClaim) }}
+  volumeClaimTemplates:
+  - metadata:
+      name: {{ template "sonarqube.fullname" . }}
+      labels:
+        release: {{ .Release.Name }}
+        chart: "{{ .Chart.Name }}"
+        app: "{{ template "sonarqube.fullname" . }}"
+    {{- with .Values.persistence.annotations  }}
+      annotations:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+    spec:
+      accessModes:
+        - {{ .Values.persistence.accessMode | quote }}
+      resources:
+        requests:
+          storage: {{ .Values.persistence.size | quote }}
+    {{- if .Values.persistence.storageClass }}
+    {{- if (eq "-" .Values.persistence.storageClass) }}
+      storageClassName: ""
+    {{- else }}
+      storageClassName: "{{ .Values.persistence.storageClass }}"
+    {{- end }}
+    {{- end }}
+  {{- end }}
```

Comment:
The changes to this file are similar to those done in this PR: https://github.com/SonarSource/helm-chart-sonarqube/pull/96 for the sonarqube-dce chart.